### PR TITLE
feat(common): builder trap-functionality testing 🏗️

### PR DESCRIPTION
**Pull request by jahorton**
Adds unit tests for some of the features added by #7319.

It's probably not the most elegant, but due to how bash `trap`s work, it's the best approach I could discern at the moment.

@keymanapp-test-bot skip

### CODE REVIEW

**Comment by jahorton** [comment]
... of course it's after I first push the PR that I realize... I could have done that within one file instead of two.

That said, I think it's better conceptually to have the actual core test code in one file with the "triggers" defined elsewhere (and thus accessible as one-liners).  But maybe that's just me.

File: resources/build/tests/trap-test-builds.sh
Comment (jahorton) [inline resources/build/tests/trap-test-builds.sh:1]: Maybe I should rename this file to `trap-test-utils.sh`?  It's meant as a utility script to provide simple 'reproductions' of scenarios to test in the actual test script `build-utils-traps.test.sh`.

File: resources/build/tests/build-utils-traps.test.sh
Comment (mcdurdin) [inline resources/build/tests/build-utils-traps.test.sh:16]:
```suggestion
CHECK="${COLOR_GREEN}✔${COLOR_RESET}"
CROSS="${COLOR_RED}❌${COLOR_RESET}"
```

Any reason we can't just do this?

**Review by mcdurdin — APPROVED** [review]
I think this is okay. Agree that the test scripts are a little clunky but if they do the job (to help us make sure we don't break builder functionality in future updates), then they'll do.

Just making sure that we aren't running the risk of trouble with nested script calls with any of this?

File: resources/build/tests/build-utils-traps.test.sh
Comment (jahorton) [reply resources/build/tests/build-utils-traps.test.sh:16]: It does look fine in Windows/MINGW64; guess I should've tried it out first.  Interestingly, the references I found when I went looking for how to do this all just used the coded form... maybe it would've been a problem in the past?

**Comment by jahorton** [comment]
> Just making sure that we aren't running the risk of trouble with nested script calls with any of this?

I'm not entirely sure what you're asking.  If it's "are we in danger of accidentally making an infinite loop", I never ran into that scenario while developing these tests.

**Comment by jahorton** [comment]
I see that I should probably hold off on merging these until after #7333 is merged.

I did use these tests as part of merge conflict resolution for #7333 (via a `temp` merge of the two), and that required renames to match its changes.  Nothing egregious, at least.

**Comment by jahorton** [comment]
I've now merged in the changes from #7333 and made it the base PR.

**Review by mcdurdin — APPROVED** [review]

**Comment by keyman-server** [comment]
Changes in this pull request will be available for download in [Keyman version 16.0.69-alpha](https://keyman.com/downloads/releases/alpha/16.0.69)

**Pull request merged**
